docs/src/real_world.md

```diff
@@ -529,6 +529,92 @@ julia> expensive_computation(5)
+
+!!! note
+    The example below assumes your using the [Plots.jl](https://github.com/JuliaPlots/Plots.jl) package and for the most part should work with any backend. The `Plots.savefig` function handles the image type based on the extension used, for example, `savefig(example,"example.png")`
+
+```julia
+using DrWatson
+using Plots
+DrWatson._wsave(filename,plot::T) where T<: Plots.Plot = begin
+    mkpath(dirname(filename));
+    savefig(plot,filename)
+end
+```
+The `filename` can be provided by [`savename`](@ref) to assist with adding parameters used to generate the data returned by [`collect_results`](@ref). Nothing new really here, but this raises the question, what if we want to get information from [`tagsave`](@ref) which provides critical details from `git` and the macro [`@tagsave`](@ref) gives us information about the script that ran and line number. We need to then work with text based plots.
+
+### Adding provenance details to text based plots
+ There are many instances where you will want to know what script generated a plot. This is hard to do consistently with plots stored in formats like `pdf` or `png`, but text-based vector formats like [`PGF/tikz`](https://en.wikipedia.org/wiki/PGF/TikZ) (or [`svg`](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics)) are just mark-up text and allow for in-file comments. Therefore, one can store comments in the actual plot file providing provenance information which enables reproducing plots/figures more easily. To do this in `DrWatson`, we have to create a new [`tagsave`](@ref) method for our data type and `Dict` of `info` we want to write. Below is an example:
+ 
+!!! note
+    The example below assumes you've add the [`PGFPlotsX.jl`](https://kristofferc.github.io/PGFPlotsX.jl/stable/) backend for [`Plots.jl`](https://github.com/JuliaPlots/Plots.jl) but this can easily be adapted for other plotting packages.
+
+```julia
+using DrWatson
+using Plots; pgfplotsx()
+
+# Puts info as comments at eof
+function insert_latex_comment(file, string::String)
+    open(file, "r+") do io
+        readlines(io)
+        @assert eof(io)
+        mark(io)
+        buf = IOBuffer()
+        seekstart(buf)
+        reset(io)
+        write(io, "% "*strip(string,'%')*"\n";)
+    end
+end
+
+DrWatson._wsave(filename, plot::T, info::Dict) where T<:Plots.Plot  = begin
+    mkpath(dirname(filename));
+    savefig(plot,filename)
+    if info["plotbackend"] == Plots.PGFPlotsXBackend
+        for (k,v) in info
+            outstr = "$k => $v"
+            insert_latex_comment(filename,outstr)
+        end
+    end
+end
+
+DrWatson.safesave(file, plot::T, info::Dict; kwargs...) where T<:Plots.Plot = begin
+    recursively_clear_path(file)
+    wsave(file,plot,info;kwargs...)
+end
```

Comment:
Wait, the point is that you are supposed to get `safesave` for free. Why is it that you need to code this? Is it because the original `safesave` method doesn't have `args...`? If so, just add `args...` to original function and we're good?